common/src/chain/tokens/rpc.rs

```diff
@@ -201,19 +161,9 @@ impl RPCNonFungibleTokenInfo {
     }
 }
 
-#[derive(Debug, Clone, Encode, Decode, serde::Serialize, serde::Deserialize, HasValueHint)]
-pub struct RPCTokenCreator(Vec<u8>);
-
-impl From<&TokenCreator> for RPCTokenCreator {
-    fn from(creator: &TokenCreator) -> Self {
-        // None-RPC type mustn't have serde requirements
-        RPCTokenCreator(creator.encode())
-    }
-}
-
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, HasValueHint)]
 pub struct RPCNonFungibleTokenMetadata {
-    pub creator: Option<RPCTokenCreator>,
+    pub creator: Option<RpcHexString>,
```

Comment:
Maybe this can be RpcAddress<Destination> as the creator is a public key?

Reply:
> Maybe this can be RpcAddress as the creator is a public key?

I don't think `PublicKey` implements `Addressable`. The closest thing that does is `Destination`.

Maybe there could be `PublicKey` directly instead of `RpcHexString`, I think it serializes to a hex string too. Would also remove the need for some conversions.